presto-main/src/test/java/io/prestosql/operator/TestGroupedTopNRowNumberAccumulator.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import io.prestosql.array.LongBigArray;
+import io.prestosql.operator.GroupedTopNRowNumberAccumulator.RowReference;
+import it.unimi.dsi.fastutil.longs.LongArrayList;
+import it.unimi.dsi.fastutil.longs.LongArraySet;
+import it.unimi.dsi.fastutil.longs.LongComparator;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestGroupedTopNRowNumberAccumulator
+{
+    @Test
+    public void testSingleGroupTopN1()
+    {
+        int topN = 1;
+        List<Long> evicted = new LongArrayList();
+        GroupedTopNRowNumberAccumulator accumulator = new GroupedTopNRowNumberAccumulator(Long::compare, topN, evicted::add);
+
+        TestingRowReference rowReference = new TestingRowReference();
+        rowReference.setRowId(0);
+        assertTrue(accumulator.add(0, rowReference));
+        assertTrue(rowReference.isRowIdExtracted());
+        assertTrue(evicted.isEmpty());
+
+        rowReference.setRowId(1);
```

Comment:
maybe add a comment like: add with same group id and larger value, which should be ignored because max is 1.  Same for others